src/engine.rs

```diff
@@ -475,8 +491,15 @@ impl Id {
 }
 
 impl BindMap {
-    fn insert_buf(&mut self, id: Id, buf: Buffer) {
-        self.buf_map.insert(id, buf);
+    fn insert_buf(&mut self, proxy: &BufProxy, buffer: Buffer) {
+        self.buf_map.insert(
+            proxy.id,
+            BindMapBuffer {
+                buffer,
+                #[cfg(feature = "buffer_labels")]
```

Comment:
This does need a None case (or taking the field out of BindMapBuffer, which might make more sense) if the cfg is not enabled.

Reply:
I originally intended to not have the field, but that meant the `buffer_labels` feature propogated very far.

The only real cost to having the names is the buffer reuse, as far as I know; the binary size for the string literals is negligible. I therefore chose to keep the names in `BufProxy` and `BindMapBuffer` in all cases, for ease of maintenance

Reply:
If the cost is low, I don't think it justifies having the feature, I'd be ok with having it on.